### DISCUSSION TITLE
How to add a rule where it checks boolean input body parameters

### DISCUSSION THREAD

**Discussion by parulbhagat-soprabanking** [mockoon/mockoon#1033] [Help]
Hi Team, 
Currently when we add a rule to check the body parameter value, then it is considering the value string "true" and boolean true as equal. How can I add a rule where it checks that the mocks return the response only if the input parameter value passed is boolean true and not when true is passed as a string like "true".
![image](https://user-images.githubusercontent.com/131943336/236438706-100e95d2-f0f8-47cc-b2c9-a06d7d440287.png)

**Comment by 255kb** [comment]
Hi, 

It's true that the rule is comparing strings to booleans but it's working. 
If you define a rule as shown in your screenshot, it will correctly compare to the boolean in the body (`{ "uncleared": true }`) and serve the response or not. 
Or maybe there is something I misunderstood in your question?

**Reply by parulbhagat-soprabanking** [reply]
Yes the screenshot that I shared does work fine with the body input { "uncleared": true } but it also give the response for { "uncleared": "true" } . What I want to implement in the rule is that it gives response for only  { "uncleared": true }  and not  { "uncleared": "true" }

**Reply by 255kb** [reply]
OK, I understand now. 
This "equals" rule is permissive by design, so it can work with such strings `"true"` or `"false"`.
We would have to add a new "boolean" rule, maybe with a switch, that would compare the type too. 
It's planned, but if you want you can propose this in a new issue.

**Reply by parulbhagat-soprabanking** [reply]
okay, thanks ! I will create an issue with this discussion just to keep a track of the issue if it is ok? Also , in the plan by when will this feature be available ?

**Reply by 255kb** [reply]
Sure you can. 
But there is no guarantee that it will be done soon. 

Do not hesitate to work on this and submit a PR. Any help is appreciated.
If you are working for a company and need this feature, we can discuss sponsoring/enterprise plans if you are interested.
